docs/docsite/rst/dev_guide/debugging.rst

```diff
@@ -0,0 +1,150 @@
+.. _debugging:
+
+=================
+Debugging Modules
+=================
+
+Debugging (local)
+=================
+
+To break into a module running on ``localhost`` and step through with the debugger:
+
+- Set a breakpoint in the module: ``import pdb; pdb.set_trace()``
+- Run the module on the local machine: ``$ python -m pdb ./my_new_test_module.py ./args.json``
+
+Debugging (remote)
+==================
+
+To debug a module running on a remote target (i.e. not ``localhost``):
+
+- On your controller machine (running Ansible) set ``ANSIBLE_KEEP_REMOTE_FILES=1`` (this tells Ansible to retain the modules it sends to the remote machine instead of removing them)
```

Comment:
Hey, it turns out rST **does** support automatically numbered lists. Instead of `1.` and `2.`, you put `#.` I tested this on a single page and am going back now to implement this across the dev guide. See https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#list-and-bullets.

Reply:
does it render on github? iirc that is main reason people stuck to 'direct numbering'